src/license.rs

```diff
@@ -4,21 +4,41 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
+use dusk_bytes::Serializable;
 use dusk_jubjub::JubJubAffine;
 use dusk_jubjub::{dhke, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED};
+use dusk_merkle::poseidon::Tree;
+use dusk_merkle::{poseidon::Item, poseidon::Opening, Aggregate};
 use dusk_pki::{PublicKey, PublicSpendKey, SecretKey, SecretSpendKey, StealthAddress};
 use dusk_poseidon::cipher::PoseidonCipher;
 use dusk_poseidon::sponge;
 use dusk_schnorr::Signature;
 use rand_core::{CryptoRng, RngCore};
 
-use dusk_bytes::Serializable;
-
 use dusk_plonk::prelude::*;
-use dusk_poseidon::tree::PoseidonBranch;
 
 use crate::state::State;
 
+pub const DEPTH: usize = 17;
+pub const ARITY: usize = 4;
+
+#[derive(Default, Debug, Clone, Copy)]
+pub struct Unit;
+
+impl<const H: usize, const A: usize> Aggregate<H, A> for Unit {
+    const EMPTY_SUBTREES: [Self; H] = [Unit; H];
+
+    fn aggregate<'a, I>(_items: I) -> Self
+    where
+        Self: 'a,
+        I: Iterator<Item = &'a Self>,
+    {
+        Unit
+    }
+}
+
+type PoseidonItem = Item<Unit>;
```

Comment:
I would prefer to have the declaration of the struct `Unit` and its `impl` into `state.rs`, as it's something related to the state.